Update transaction-life-cycle.adoc with finality status of deploy_account and declare transactions (#1295)
starknet-io/starknet-docs: components/Starknet/modules/architecture-and-concepts/pages/network-architecture/transaction-life-cycle.adoc

[NOTE]
====
A rejected transaction is stored in the mempool. You cannot send another transaction with the same transaction hash.
====
|`ACCEPTED_ON_L2` |The transaction was executed and entered an actual created block on L2.
|`ACCEPTED_ON_L1` |The transaction was accepted on Ethereum.

> .2+|*Execution* a|`REVERTED` |The transaction passed validation but failed during execution in the sequencer. It is included in the block with the status `REVERTED`.

Reviewer: ```suggestion
.2+|*Execution* |`REVERTED` a|The transaction passed validation but failed during execution in the sequencer. It is included in the block with the status `REVERTED`.
```

[NOTE]
====

> Since only `INVOKE` transactions have an execution phase, `DEPLOY_ACCOUNT` and `DECLARE` transactions cannot be reverted. If the `+__VALIDATE_DEPLOY__+` or `+__VALIDATE_DECLARE__+` function, respectively, fail when run in the sequencer, then the transaction will be rejected.

Reviewer: ```suggestion
Since only `INVOKE` transactions have an execution phase, `DEPLOY_ACCOUNT` and `DECLARE` transactions cannot be reverted. If either the `+__VALIDATE_DEPLOY__+` or the `+__VALIDATE_DECLARE__+` function fails when run in the sequencer, then the transaction is rejected.
```

|`ACCEPTED_ON_L2` |The transaction was executed and entered an actual created block on L2.
|`ACCEPTED_ON_L1` |The transaction was accepted on Ethereum.
.2+|*Execution* |`REVERTED` |The transaction passed validation but failed during execution in the sequencer. It is included in the block with the status `REVERTED`.

> Note: Since only INVOKE transactions have an execution phase, DEPLOY_ACCOUNT and DECLARE transactions cannot be reverted. They can be `REJECTED` if `+__VALIDATE_DEPLOY__+` or `+__VALIDATE_DECLARE__+`, respectively, fail when run in the sequencer. 

Author: @stoobie Can you please help to make this a note similar to the one in the rejected cell in the table?
